rootchain/sampool/bucket.go

```diff
@@ -0,0 +1,63 @@
+package sampool
+
+import (
+	"github.com/0xPolygon/polygon-edge/rootchain"
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+type samSet struct {
+	messages   []rootchain.SAM
+	signatures map[string]bool
+}
+
+func newUniqueSAMs() samSet {
+	return samSet{
+		messages:   make([]rootchain.SAM, 0),
+		signatures: make(map[string]bool),
+	}
+}
+
+func (s *samSet) add(msg rootchain.SAM) {
+	strSignature := string(msg.Signature)
```

Comment:
How about converting bytes to hex for easier debug

Reply:
Do we have a helper utility for that?

Reply:
hex package has

Reply:
[Fixed](https://github.com/0xPolygon/polygon-edge/pull/718/commits/4de7093b9051caaff411b7743d30395affb64ad1)